### DISCUSSION TITLE
Vite:build is not working but vite:dev is

### DISCUSSION THREAD

**Discussion by pappuchai** [remix-run/remix#9312] [Q&A] [answered]
The below route structure using **[remix-flat-routes](https://github.com/kiliman/remix-flat-routes)** is working fine with vite:dev but the page goes blank with vite:build.   I can see the page render for a fraction of a second and the page goes blank. 

<img width="254" alt="Screenshot 2024-04-25 at 12 05 18 PM" src="https://github.com/remix-run/remix/assets/168064951/3a761afe-ffeb-43e4-9d7c-c0ca6109f604">   

My Vite.config.ts is this
<img width="524" alt="Screenshot 2024-04-25 at 12 11 14 PM" src="https://github.com/remix-run/remix/assets/168064951/a08ad061-8815-4bcd-80ea-81c1750441c6">

Home page goes blank after a fraction of second with minified react error (not useful) and on  http:/localhost:3000/blogs in the URL directly,  I see the following 404 error
<img width="1434" alt="Screenshot 2024-04-25 at 12 12 36 PM" src="https://github.com/remix-run/remix/assets/168064951/31f2cc73-9786-409a-b8a5-4d08e4dc9807">

Remix-dev-tools on dev mode doesn't show any hydration errors or other errors.

**Another project with a similar structure but without vite works pefectly with dev and build.**  

Can someone help where this is going wrong ?

**Comment by pappuchai** [comment]
Update:  Looks like hydration issue.  But happens only on vite:build and not on vite:dev.   Trying to go through to find the cause.  Suggestions welcome.  

Update #2:  Removed remix-flat-routes and did with native remix route conventions.   vite:build is still not working,  console shows "hydration" error

**Comment by ronalterya** [comment]
Problem with rollup 4.16. pin to rollup 4.15.

**Comment by pappuchai** [comment, marked as answer]
Updated React to 18.3 canary version to prevent hydration issues.    As per this [repo](https://github.com/kiliman/remix-vite-template)
